plonky2/src/iop/challenger.rs

```diff
@@ -143,7 +151,9 @@ impl<F: RichField, H: Hasher<F>> Challenger<F, H> {
             self.sponge_state = H::Permutation::permute(self.sponge_state);
         }
 
-        self.output_buffer = self.sponge_state[0..SPONGE_RATE].to_vec();
+        self.output_buffer.clear();
```

Comment:
Is this to avoid an allocation? Maybe we can do just in the (more performance sensitive) native verifier since it's sligthly more verbose.

Reply:
Yes, it's to avoid the allocation.

Reply:
we can also make a helper function if we want to improve readability.